### PR TITLE
[release-4.13] OCPBUGS-24391: Limit OVN-Kubernetes RBAC permissions

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
in addition to https://github.com/openshift/cluster-network-operator/pull/2093 this will limit RBAC permissions of ovn-kubernetes as we have done upstream 

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: dist/images/daemonset.sh [hunk 1]
```diff
@@ -600,6 +600,10 @@ net_cidr=${net_cidr} svc_cidr=${svc_cidr} \
   in_upgrade=${in_upgrade} \
   j2 ../templates/ovn-setup.yaml.j2 -o ${output_dir}/ovn-setup.yaml
 
+## interconnect features removed from rbac-ovnkube-node.yaml
```
Comment (jcaamano) [inline dist/images/daemonset.sh:603]: nit: I wouldn't mention interconnect if there is no interconnect, can lead to confusion

File: dist/images/daemonset.sh [hunk 2]
```diff
@@ -600,6 +600,10 @@ net_cidr=${net_cidr} svc_cidr=${svc_cidr} \
...
+cp ../templates/rbac-ovnkube-node.yaml.j2 ${output_dir}/rbac-ovnkube-node.yaml
+cp ../templates/rbac-ovnkube-master.yaml.j2 ${output_dir}/rbac-ovnkube-master.yaml
+cp ../templates/rbac-ovnkube-db.yaml.j2 ${output_dir}/rbac-ovnkube-db.yaml
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply dist/images/daemonset.sh:606]: yeah that went in with the cno repo PR

File: dist/images/daemonset.sh (hunk 1 above)
Comment (JacobTanenbaum) [reply dist/images/daemonset.sh]: done

File: dist/images/daemonset.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (jcaamano) [reply dist/images/daemonset.sh:605]: but don't you need them here as well for the kind environment?

File: go-controller/pkg/clustermanager/network_cluster_controller.go [hunk 3]
```diff
@@ -302,8 +302,10 @@ func (ncc *networkClusterController) updateNodeSubnetAnnotationWithRetry(nodeNam
 				return fmt.Errorf("failed to update node %q annotation subnet %s",
 					node.Name, util.JoinIPNets(hostSubnets, ","))
 			}
+			// It is possible to update the node annotations using status subresource
+			// because changes to metadata via status subresource are not restricted for nodes
```
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply go-controller/pkg/clustermanager/network_cluster_controller.go]: done

File: dist/images/daemonset.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (JacobTanenbaum) [reply dist/images/daemonset.sh:605]: accidentally forgot to git add those files

File: dist/templates/rbac-ovnkube-node.yaml.j2 [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+roleRef:
+    name: ovn-k8s-configmap
+    kind: Role
+    apiGroup: rbac.authorization.k8s.io
+subjects:
+    - kind: ServiceAccount
+      name: ovnkube-node
+      namespace: ovn-kubernetes
+
+# In IC ovnkube-node pod needs configmap access in ovn-k ns for topology version:
```
Comment (kyrtapz) [inline dist/templates/rbac-ovnkube-node.yaml.j2:36]: NIT: I think it means there is no need for the `ovnkube-node-configmaps` RoleBinding in 4.13.
We can probably remove it together with this comment.

File: dist/templates/rbac-ovnkube-node.yaml.j2 [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      resources:
+          - events
+      verbs: ["create", "patch", "update"]
+    - apiGroups: [""]
+      resources:
+          - nodes/status
+      verbs: [ "patch", "update" ]
+
+# Without IC endpoints are read by ovnkube-node on startup
+# With IC endpoints are created by ovnkube-zone-controller/sb-ovsdb startup script in multinode-zone for IC
```
Comment (kyrtapz) [inline dist/templates/rbac-ovnkube-node.yaml.j2:80]: NIT: Lets remove mentions about IC from the comment

File: dist/templates/rbac-ovnkube-node.yaml.j2 (hunk 4 above)
Comment (JacobTanenbaum) [reply dist/templates/rbac-ovnkube-node.yaml.j2]: done

**Comment by kyrtapz** [comment]
/retitle [release-4.13] OCPBUGS-24391: Limit OVN-Kubernetes RBAC permissions

**Pull request merged**